Mini-cart: add setting to not render the block on the cart & checkout pages

albarin opened the pull request:
Currently, the `Mini Cart` is hidden with CSS on the Cart & Checkout pages. As discussed on https://github.com/woocommerce/woocommerce-blocks/issues/7760 we want to give the users the option to not render it at all (not only hide it).
This PR add this setting to the `Mini Cart` block and does not render the block when enabled. The default setting is to render it invisible (same behavior as until now) to not affect existing stores.

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/7760
### Screenshots

### Testing
#### User-Facing Testing

**Site Editor**
1. Go to the Site Editor > Template parts and edit the header template to add the `Mini Cart` block. Save.
2. Edit the block and make sure you see the new setting:

<img width="282" alt="Screenshot 2023-03-13 at 16 27 26" src="https://user-images.githubusercontent.com/186112/224748404-6c02b9ef-2112-405b-9ab5-69abe857a03c.png">

3. Make sure the default option is `Hide`.
4. In the store, go to the Cart page and make sure the `Mini Cart` is rendered but invisible. Repeat but for the Checkout page.
5. Go back to the Site Editor, change the `Mini Cart` setting to `Remove`, and save.
6. In the store, go to the Cart page and make sure the `Mini Cart` markup is not rendered at all. Repeat but for the Checkout page.


**Post/page**
1. Create a new post or page.
2. Insert the `Mini Cart`.
3. Make sure the new `Mini Cart in cart and checkout pages` setting does not appear.

### WooCommerce Visibility
* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog
> Mini Cart: add a new setting to allow not rendering the block on the cart & checkout pages.

albarin committed:
Add `renderInCartAndCheckout` attribute and toggle

albarin committed:
Show setting only when inserted in a template

albarin committed:
Don't render the mini-cart if the setting is disabled

albarin committed:
Stop rendering the mini-cart if the setting is false

albarin commented:
@pmcpinto @Aljullu do you have any suggestions for the toggle options and description copy?

albarin committed:
Merge branch 'trunk' into fix/7760-mini-cart-checkout-page

Aljullu commented on src/BlockTypes/MiniCart.php:
```diff
@@ -567,4 +571,15 @@ public function register_empty_cart_message_block_pattern() {
+	 *
+	 * @return bool
+	 */
+	public function shouldNotRenderMiniCart( array $attributes ) {
```
Should we use snake case for PHP methods?

albarin replied:
True, fixed 🙏

Aljullu commented on assets/js/blocks/mini-cart/index.tsx:
```diff
@@ -61,6 +61,10 @@ const settings: BlockConfiguration = {
 			type: 'boolean',
 			default: false,
 		},
+		makeInvisibleInCartAndCheckout: {
+			type: 'boolean',
+			default: true,
+		},
```
I'm not sure about this, but given the discussion in #7760, I wonder if we should make this a `string` so it can support more options in the future. (I know we don't have any plans right now to add more options, but in case some day we do, it would make our lives easier than if it's a boolean, what do you think?) :thinking:

albarin replied:
Yeah, I was in doubt about it as well, I think you are right, it makes sense to make it more future-proof, I'll change it.

Aljullu reviewed (commented):
I left a couple of comments in the code, but overall LGTM. I also tested it and it works perfectly. :ok_hand: 

> @pmcpinto @Aljullu do you have any suggestions for the toggle options and description copy?

It's a tricky one. An alternative would be to name the two options as "invisible" and "hidden", but I don't know if it makes things any easier to understand. Or to replace `Don't render` with `Don't display`.

Another option would be to use the explanation text below the toggle to explain what each options does. We seem to do it in some blocks, but I'm not sure if it's a correct GB pattern. :confused: 

[Enregistrament de pantalla des de 2023-03-10 14-13-45.webm](https://user-images.githubusercontent.com/3616980/224325623-64df4926-8f46-4484-b099-288dde628a33.webm)

albarin committed:
method to snakecase

albarin commented:
> It's a tricky one. An alternative would be to name the two options as "invisible" and "hidden", but I don't know if it makes things any easier to understand. Or to replace `Don't render` with `Don't display`.

More thoughts: what about `Hide` and `Remove`? 🤔  

> Another option would be to use the explanation text below the toggle to explain what each options does. We seem to do it in some blocks, but I'm not sure if it's a correct GB pattern. 😕

Yeah, that'd be useful, but I remember I changed that behavior some time ago on another toggle, so maybe it's something we are trying to avoid, not sure;

albarin committed:
Make the attribute a string

In the future we may want to add more rendering styles, so a boolean wasn't the
best option, a string will allowed us more than two options

pmcpinto commented:
> More thoughts: what about Hide and Remove? 🤔

 I was going to suggest this as an alternative :)

albarin committed:
Change toggle copy

Aljullu reviewed (approved):
> More thoughts: what about `Hide` and `Remove`? 🤔  

I like it! :ok_hand: 

PR looks good to me. :shipit:
